test/integration/kubernetes_deploy_test.rb

```diff
@@ -632,6 +632,18 @@ def test_can_deploy_deployment_with_zero_replicas
     ])
   end
 
+  def test_can_deploy_deployment_with_partial_rollout_success
```

Comment:
nit: the point is more that a deployment with the maxUnavailable required-rollout annotation does not wait for full availability (and the copy in `restart_task_test` should talk about restarts)